x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_settings/conversation_settings.tsx

```diff
@@ -131,18 +139,21 @@ export const ConversationSettings: React.FC<ConversationSettingsProps> = React.m
       [selectedConversation, setUpdatedConversationSettings]
     );
 
-    const selectedConnectorId = useMemo(
-      () => selectedConversation?.apiConfig.connectorId,
-      [selectedConversation?.apiConfig.connectorId]
-    );
+    const selectedConnector = useMemo(() => {
+      const selectedConnectorId = selectedConversation?.apiConfig.connectorId;
+      if (areConnectorsFetched) {
+        return connectors?.find((c) => c.id === selectedConnectorId);
+      }
+      return undefined;
+    }, [areConnectorsFetched, connectors, selectedConversation?.apiConfig.connectorId]);
 
     const selectedProvider = useMemo(
       () => selectedConversation?.apiConfig.provider,
       [selectedConversation?.apiConfig.provider]
     );
 
     const handleOnConnectorSelectionChange = useCallback(
-      (connectorId: string, provider: OpenAiProviderType) => {
+      (connectorId: string, provider?: OpenAiProviderType, model?: string) => {
```

Comment:
Multiple optional params no bueno. Will refactor to just use connector itself as that will be cleaner 👍

Reply:
Cleaned up to just pass along the `connector` as a whole since it contains additional useful context (e.g. `isPreconfigured`)